src/components/List/__tests__/List.spec.tsx

```diff
@@ -0,0 +1,215 @@
+        beforeEach(() => {
+            view = render(
+                <List>
+                    {itemNumbersArray.map((number) => (
+                        <ListItem>Item ${number}</ListItem>
+                    ))}
+                </List>
+            );
+            listItems = screen.getAllByRole('option');
+        });
+
+        it('should not have any item highlighted from the start', () => {
+            expect(view.container).toMatchSnapshot();
+            itemNumbersArray.forEach((number) => {
+                expect(getListItemAt(number)).not.toHaveAttribute('isHighlighted');
+            });
+        });
+
+        it('should move highlight on components in both directions properly', async () => {
+            const user = userEvent.setup();
+            expect(view.container).toMatchSnapshot();
+            await user.click(listItems[0]);
+            await navigateDown();
+
+            expect(screen.getAllByRole('option')[0]).toHaveAttribute('aria-selected');
+
+            await navigateDown();
+
+            expect(getListItemAt(0)).toHaveAttribute('aria-selected', 'false');
+            expect(getListItemAt(1)).toHaveAttribute('aria-selected', 'true');
+
+            await navigateUp();
+
+            expect(getListItemAt(0)).toHaveAttribute('aria-selected', 'true');
+            expect(getListItemAt(1)).toHaveAttribute('aria-selected', 'false');
+        });
+
+        it('should not let highlighted item got out of list bounds', async () => {
+            const user = userEvent.setup();
+            await user.click(listItems[0]);
+            await navigateUp();
+            await navigateUp();
+
+            expect(getListItemAt(0)).toHaveAttribute('aria-selected', 'true');
+
+            await navigateDown();
+            await navigateDown();
+            await navigateDown();
+            await navigateDown();
+            await navigateDown();
+
+            expect(getListItemAt(itemNumbersArray.length - 1)).toHaveAttribute(
+                'aria-selected',
+                'true'
+            );
+        });
+    });
+
+    describe('Callbacks', () => {
+        beforeEach(() => {
+            view = render(
+                <List>
+                    {itemNumbersArray.map((number) => (
+                        <ListItem onClick={() => mockOnClick(number)}>Item ${number}</ListItem>
+                    ))}
+                </List>
+            );
+            listItems = screen.getAllByRole('option');
+        });
+
+        it('should call onClick after selecting the highlighted item', async () => {
+            const user = userEvent.setup();
+            await user.click(listItems[0]);
+
+            await navigateDown();
+            await navigateDown();
+            await navigateDown();
+            await userEvent.keyboard('[Enter]');
+
+            expect(mockOnClick).toBeCalledWith(2);
+        });
+        it.skip('should not call onClick after navigating to the next highlighted item', async () => {
```

Comment:
I'm a bit stuck with this tests case. 
Here we click on element and then expect that Enter will not clicked or first click doesn't count ?

Reply:
I tried to fix it, but indeed took me quite a bit of time to figure out what these tests should do, and why they don't really work. While investigating, I ended up refactoring quite a bit if the file, adding some tests, and deleting others. I think the suit looks better now (but I'm biased 😝). Since there were so many changes, I thought it is easier to just push my changes instead of trying to describe them here. It is your PR, if you don't like my changes, you can revert them and apply only the parts you think make sense.

Reply:
I've checked it looks great